torch/export/exported_program.py

```diff
@@ -1161,6 +1161,28 @@ def _update(
             verifiers=verifiers if verifiers is not None else self.verifiers,
         )
 
+    def _move_to(self, location: Union[torch.device, str, Dict[str, str]]):
```

Comment:
per https://docs.google.com/document/d/1DCp9qdzgoLFLPbe6KpQ1JfKT_7cTpAzt-5OccuPvOgg/edit?usp=sharing

you will also need to handle device in tensor's metadata. 

Also, let's talk about how do we expect user to use this?